docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+
+#### Stalling revealed key propagation via turbine by leaders
+
+Naturally, leaders want their blocks to be included in the major fork. However,
+malice leader can still annoy the cluster by stalling the descrambling phase
+in this way. To mitigate this, we allow subsequent leaders can also propagate
+the revealed keys shreds.
+
+#### Selfish mining by only including theirs own MEV transactions
+
+the node must be smarter than all, otherwise reward will be less.
+
+also, may impose 1/3+ vote instruction in block not to be skipped.
+
+#### Spamming with garbage scrambles by leader or (malicious) validators
+
+- generally should be fine. the leader just caps the max txes in a block
+ - mining bad scramble reward less or none? (forced and chunked descrabmling will be needed for proper rewarding system in that case)
+
+#### Stalling the banking stage after full revelation of scrambled transactions
+
+Likewise above, the mitigation here is that the next leader can replay the
+previous leader's block if the previous leader has silenced.
+
+### Optimal behaviors for each role
+
+#### Clients
+
+They don't want to be targeted by MEV for any trading transactions.
+Thereby they scramble the transaction and never share the derived private key to
+descramble. Also, they can't withhold revelation of their scrambled transaction
+as its outcome turned out unfavorable. Also, they want to protect their
+transactions from cluster stalls. So, they make transactions quickly expire.
+Thus, their strategy for most likely favorable trade is submission of their
+transaction as fast as possible as determined.
+
+Lastly, transactions are now included into a block very greedily, meaning the
+5000 lamports fee will be debited once broadcast (and included into a block).
+And observation by the cluster can be signalled very timely and very reliably
+to that client. So, clients (esp.  bots) are less incentivised to spamming for
+the race of inclusion.
+
+#### Leaders
+
+They want to maximize profit from running the validator in any possible ways.
+So, they try to stall the block production for MEV with the help of searchers.
+However, stalling block production is guaranteed to result in skipped block. All
+in all, this behavior incurs opportunity cost. Thus, their optimal behavior is
+include transactions indiscriminately as many as possible for maximum block
+rewards. This naturally results in the FIFO ordering of transactions.
+
+#### Validators
+
+They want cluster fork less frequently so generally cooperates.
+Unless significant nodes (like more than 1/3) are colluding, they pre-vote
+blocks honestly with accurate wall clock in the fear of reduced rewards because
+of being outstanding outlier otherwise.
+
+Also they're incentivised for publishing scrambled transactions as first solver
+by small rewards. Lying the results is eligible for slashing to ensure bad
+scrambled transaction is detected and backed by stakes.
```

Comment:
Maybe clarify what exactly validator could be lying about (how bad scrambled transaction would look like), and how slashing is supposed to work (who is gonna monitor for those bad transactions, and what happens after).

Reply:
the tx-submitting honest user will monitors whether their tx is correctly descrabmled. if not (i.e. the transaction is errorneously marked as garbage), user submits another tx with fraud proof with original scrambled tx's signature and correct descrambling key). 

after quick verification (just descrambe and see proper sig inside it), every leader's pubkey who claimed that the tx is garbage can be slashed.